lldb-eval/eval.cc

```diff
@@ -981,7 +981,10 @@ Value Interpreter::EvaluateComparison(BinaryOpKind kind, Value lhs, Value rhs) {
   }
 
   // Must be pointer/integer and/or nullptr comparison.
-  bool ret = Compare(kind, lhs.GetUInt64(), rhs.GetUInt64());
+  size_t ptr_size = target_.GetAddressByteSize();
+
+  bool ret = Compare(kind, lhs.GetInteger().trunc(ptr_size),
+                     rhs.GetInteger().trunc(ptr_size));
```

Comment:
I think `APInt::trunc(width)` takes the width in _bits_ -- https://github.com/llvm/llvm-project/blob/6cac3dc9db57cbf5f9f9092d97f878be3db5020d/llvm/include/llvm/ADT/APInt.h#L1242
This probably works by accident :P